internal/core/command/config/config.go

```diff
@@ -34,6 +35,29 @@ type WritableInfo struct {
 	InsecureSecrets bootstrapConfig.InsecureSecrets
 }
 
+type MessageQueue struct {
+	Internal InternalMessageQueue
+	External ExternalMQTT
+}
+
+type InternalMessageQueue struct {
+	bootstrapConfig.MessageBusInfo
```

Comment:
Inline structs do not work with our consul implementation and must be avoided.
```suggestion
	MessageBusInfo bootstrapConfig.MessageBusInfo
```

Reply:
In this case should we just copy the necessary fields from `MessageBusInfo` so that we don't need another level in configuration.toml:

```toml
[MessageQueue]
  [MessageQueue.Internal]
    [MessageQueue.Internal.MessageBusInfo]
    Type = "redis"
    Protocol = "redis"
    Host = "localhost"
    Port = 6379
    ...
```

Reply:
IMO, this extra level is ok. Eliminating it creates more complex and duplicated code

Reply:
What if we added a map for the topics to `MessageBusInfo`? Then the topics would be in the extra level instead. In 3.0 the existing `PublishTopicPrefix` and `SubscribeTopic` would be moved into the topics map. This allows `MessageBusInfo` to be more flexible with respect to topics.

```toml
[MessageQueue]
  [MessageQueue.Internal]
  Type = "redis"
  Protocol = "redis"
  Host = "localhost"
  Port = 6379
  AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
  SecretName = "redisdb"
    [MessageQueue.Internal.Topics]
    RequestTopicPrefix = "edgex/command/request/"   # for publishing requests to the device service; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
    ResponseTopic = "edgex/command/response/#"      # for subscribing to device service responses
    InternalRequestCommandTopic = "/command/request/#"     # for subscribing to internal command requests
    InternalResponseCommandTopicPrefix = "/command/response/"    # for publishing responses back to internal service /<device-name>/<command-name>/<method> will be added to this publish topic prefix
    InternalRequestQueryTopic = "/commandquery/request"   # for subscribing to internal command query requests
    InternalResponseQueryTopic = "/commandquery/response"   # for publishing reponses back to internal service
```